hardening/audit-rules-grouping/test.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/python3
+
+import re
+
+from lib import util, results, versions
+
+syscalls_groups = [
+    ['setxattr', 'lsetxattr', 'fsetxattr', 'removexattr', 'lremovexattr', 'fremovexattr'],
+    ['init_module', 'delete_module', 'query_module', 'finit_module'],
+    ['open', 'creat', 'truncate', 'ftruncate', 'openat', 'open_by_handle_at'],
+    ['lchown', 'fchown', 'chown', 'fchownat'],
+    ['adjtimex', 'settimeofday'],
+    ['clock_settime'],
+    ['sethostname', 'setdomainname'],
+]
+# fchmodat2 is only available since RHEL 10+ kernels,
+# CaC/content rule for renameat2 is only on RHEL 10+
+if versions.rhel >= 10:
+    syscalls_groups.append(['chmod', 'fchmod', 'fchmodat', 'fchmodat2'])
+    syscalls_groups.append(['unlink', 'unlinkat', 'rename', 'renameat', 'renameat2', 'rmdir'])
+else:
+    syscalls_groups.append(['chmod', 'fchmod', 'fchmodat'])
+    syscalls_groups.append(['unlink', 'unlinkat', 'rename', 'renameat', 'rmdir'])
+
+
+def syscalls_pretty_print(syscalls):
+    return ','.join(syscalls)
+
+
+def unselect_all_rules_except_audit(orig_ds, new_ds):
+    rule_line = re.compile(r'<.*Rule.*id="xccdf_org.ssgproject.content_rule_')
+    audit_rules = re.compile(
+        r'<.*Rule.*id="xccdf_org.ssgproject.content_rule_audit_rules.*'
+        r'(_unsuccessful_file_modification|_dac_modification|'
+        r'_file_deletion_events|_kernel_module_loading|'
+        r'_networkconfig_modification|_time_adjtimex|'
+        r'_time_clock_settime|_time_settimeofday|_time_stime)',
+    )
+
+    with open(orig_ds) as orig_ds_f:
+        with open(new_ds, 'w') as new_ds_f:
+            for line in orig_ds_f:
+                if rule_line.search(line):
+                    if audit_rules.search(line):
+                        line = line.replace('selected="false"', 'selected="true"')
+                    else:
+                        line = line.replace('selected="true"', 'selected="false"')
+                new_ds_f.write(line)
```

Comment:
Maybe consider extending / optimizing `oscap.unselect_rules()` instead. Theoretically, you could do something like
```python
rules = list(oscap.global_ds().rules.keys())
oscap.unselect_rules(old_ds, new_ds, rules)
```
with the current `lib/` code already, but `oscap.unselect_rules()` tries to compile a regexp for every single rule and maybe that could be optimized by searches for XML structures that look like rule definitions + a second `in` check for each of the rule names.

Reply:
After unselecting all rules we would need to select the audit rules again so there would be more changes needed and we agreed it is not worth implementing for this single use case.